### ISSUE TITLE
Output to a file and get live video preview

### ISSUE THREAD

**Issue by DenisIvanovIvanov** [rosenbjerg/FFMpegCore#367]
Hello!

I just discovered this great wrapper. We have app which can record from whatever device (camera) is configured or desktop or IP camera, all depending on configuration with live preview (SDL2), using ffmpeg binaries (libav) directly. 

Works so far, but it's getting more and more complex just for the live preview basically, which introduces bugs hard to fix/debug. I've started digging if we can use ffmpeg cli directly, as is more mature, to save us the complexity of supporting many different devices and encoders. 

Tl;dr is it possible with this wrapper to record to a file and also get the frames in real time for the preview? 

That would be really great for us.

**Comment by rosenbjerg** [comment]
I think it would take a lot of work to support that in this library, and haven't heard from anyone looking into creating a PR with this. So while I'm not against supporting this, I think it is pretty far away at this point, and may never be supported directly

**Comment by rosenbjerg** [comment]
Check out [Ruslan-B/FFmpeg.AutoGen](https://github.com/Ruslan-B/FFmpeg.AutoGen) for more direct usage of ffmpeg
